src/app/services/table.service.ts

```diff
@@ -158,6 +160,25 @@ export class TableService {
     );
   }
 
+  getDimensions() {
+    const selectionConstraint = this.constraintService.generateSelectionConstraint();
+    const projectionConstraint = this.constraintService.generateProjectionConstraint();
+    let combo = new CombinationConstraint();
+    combo.addChild(selectionConstraint);
+    combo.addChild(projectionConstraint);
+    this.resourceService.getStudyNames(combo).subscribe(
```

Comment:
the study names are already obtained by in the constructor of constraint service